test/core/slice/slice_test.cc

```diff
@@ -241,9 +241,15 @@ static void test_slice_interning(void) {
   LOG_TEST_NAME("test_slice_interning");
 
   grpc_init();
-  grpc_slice src1 = grpc_slice_from_copied_string("hello123456789123456789");
-  grpc_slice src2 = grpc_slice_from_copied_string("hello123456789123456789");
+  grpc_slice src1 = grpc_slice_from_copied_string("hello1234567891234567891");
+  grpc_slice src2 = grpc_slice_from_copied_string("hello1234567891234567891");
+
+  // make sure slices are refcounted to guarantee slices' start ptrs are
+  // distinct (even on windows opt 64bit build).
+  // See https://github.com/grpc/grpc/issues/20519
+  GPR_ASSERT(src1.refcount);
   GPR_ASSERT(GRPC_SLICE_START_PTR(src1) != GRPC_SLICE_START_PTR(src2));
```

Comment:
alternatively we could add (&src1 != &src2) condition to the assert which also unbreaks the test.

Reply:
Personally I like this idea of adding `GPR_ASSERT(&src1 != src2)`, I think it makes the theory explained here more explicit.